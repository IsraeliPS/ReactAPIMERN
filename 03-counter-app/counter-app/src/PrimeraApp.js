// import React, { Fragment } from 'react'
import React from 'react'
import PropTypes from 'prop-types'

const PrimeraApp = ({saludo, subtitulo}) => {
    // const saludo ="Hola koders"
    
    return (
        <>
            <h1>{saludo}</h1>
            <p>{subtitulo}</p>
        </>
        )
}
PrimeraApp.protoTypes={
    saludo:PropTypes.string.isRequired
}
PrimeraApp.protoTypes={
    subtitulo:"subtitulo"
}
export default PrimeraApp;