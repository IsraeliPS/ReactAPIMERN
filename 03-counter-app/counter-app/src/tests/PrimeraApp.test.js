// import { render } from '@testing-library/react'

import React from 'react'

import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import PrimeraApp from '../PrimeraApp'
import '../setupTest'

describe('Pruebas en elemento react', () => {
  // test('Debe mostrar en dom Hola soy Goku ', () => {
  //     const saludo="Hola, soy Goku";

  //     const {getByText} = render( <PrimeraApp saludo={saludo}/>);
  //     expect(getByText(saludo)).toBeInTheDocument();

  // })
  test('debe mostrar <PrimeraApp /> correctamente', () => {
    const saludo = 'Hola, Soy Goku'
    const wrapper = shallow(<PrimeraApp saludo={saludo} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('debe de mostrar el subtitulo enviado por props', () => {
    const saludo = 'Hola, Soy Goku'
    const subTitulo = "i'll be back"
    const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={subTitulo} />)
    const textoParrafo = wrapper.find('p').text()
    console.log(textoParrafo)
    expect(textoParrafo).toBe(subTitulo)
  })
})
