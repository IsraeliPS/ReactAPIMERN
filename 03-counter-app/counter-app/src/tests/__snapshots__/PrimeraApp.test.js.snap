// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en elemento react debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, Soy Goku
    ¡¡¡
  </h1>
  <p />
</Fragment>
`;
